fem/fespace.hpp

```diff
@@ -92,7 +92,127 @@ class FaceQuadratureInterpolator;
+    for degrees of freedom associated with "hanging" nodes, edges, or faces to
+    be constrained by degrees of freedom associated with another mesh entity.
+    In such cases the "hanging" degrees of freedom should not be considered
+    "true" degrees of freedom since their values cannot be independently
+    assigned. For this reason the FiniteElementSpace must process these
+    constraints and define a reduced set of "true" degrees of freedom which are
+    distinct from the local degrees of freedom.
+    @par
+    The second important distinction arises in parallel processing. When
+    distributing a linear system in parallel each degree of freedom must be
+    assigned to a particular processor, its owner. From the finite element
+    point of view it is convenient to distribute a computational mesh and
+    define an owning processor for each element. Since degrees of freedom may
+    be shared between neighboring elements they may also be shared between
+    neighboring processors. Another role of the FiniteElementSpace is to
+    identify the ownership of degrees of freedom which must be shared between
+    processors. Therefore the set of "true" degrees of freedom must also remove
+    redundant degrees of freedom which are owned by other processors.
+    @par
+    To summarize the set of true degrees of freedom are those degrees of
+    freedom needed to solve a linear system representing the partial
+    differential equation being modeled. True dofs differ from "local" dofs by
+    eliminating redundancies across processor boundaries and applying
+    the constraints needed to properly define fields on non-conforming meshes.
+
+    @anchor vdof @par Vector DoF:
+    %Vector dofs or @b vdofs are related to fields which are constructed using
+    multiple copies of the same set of basis functions. A typical example would
+    be the use of three instances of the scalar H1 basis functions to
+    approximate the x, y, and z components of a displacement vector field in
+    three dimensional space as often seen in elasticity simulations.
+    @par
+    %Vector dofs do not represent a specific index space the way the three
+    previous types of dofs do. Rather they are related to modifications of
+    these other index spaces to accomodate multiple copies of the underlying
+    function spaces.
+    @par
+    When using @b vdofs, i.e. when @b vdim != 1, the FiniteElementSpace only
+    manages a single set of degrees of freedom and then uses simple rules to
+    determine the appropriate offsets into the full index spaces. Two ordering
+    rules are supported; @b byNODES and @b byVDIM. See Ordering::Type for
+    details.
+    @par
+    Clearly the notion of a @b vdof is relevant in each of the three contexts
+    mentioned above so extra care must be taken whenever @b vdim != 1 to ensure
+    that the @b edof, @b ldof, or @b tdof is being interpretted correctly.
+ */
```

Comment:
Thanks for putting these detailed descriptions into the doxygen.  It's important that we have this information here.